### ISSUE TITLE
NSD 4.3.0 does not start on FreeBSD

### ISSUE THREAD

**Issue by h-ume** [NLnetLabs/nsd#78]
NSD 4.3.0 fails to start on FreeBSD with following error:

    [2020-03-19 10:56:03.988] nsd[6411]: error: setsockopt(..., SO_SETFIB, 0, ...) failed: Bad address

The following change should fix this problem.
```
--- server.c.orig	2020-03-17 10:00:17 UTC
+++ server.c
@@ -1104,7 +1104,7 @@ set_setfib(struct nsd_socket *sock)
 {
 #if defined(SO_SETFIB)
 	if(setsockopt(sock->s, SOL_SOCKET, SO_SETFIB,
-	              (const void *)(uintptr_t)sock->fib, sizeof(sock->fib)) == -1)
+	              (const void *)&sock->fib, sizeof(sock->fib)) == -1)
 	{
 		log_msg(LOG_ERR, "setsockopt(..., %s, %d, ...) failed: %s",
 		                 "SO_SETFIB", sock->fib, strerror(errno));
```

**Comment by wcawijngaards** [comment]
Thanks for the fix and the report!

**Comment by k0ekk0ek** [comment]
My bad. Very sloppy. Thanks for the patch! Since that means you specified a specific routing table, would love to know how you're using it and if it works for you at some point (if you can spare the time of course). Thanks again.

**Comment by h-ume** [comment]
I have two routing tables.
I'm using ip-address option.
But, I do not specify setfib option in nsd.conf.

**Comment by k0ekk0ek** [comment]
Hmmm, then there's possibly another logic error in the code. The `SO_SETFIB` should only be enabled if `setfib` is specified in the configuration. I think it's when using the wildcard addresses then(?) In which case the `fib` member of the `nsd_socket` struct isn't initialized to `-1`... I'll have a look

**Comment by k0ekk0ek** [comment]
Seems like my hunch was right. Addon fix in #79. Thanks again for the report.
